pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,440 @@
 package git
 
 import (
-	"bufio"
-	"bytes"
-	"crypto/tls"
-	"crypto/x509"
 	"fmt"
-	"io"
-	"io/ioutil"
-	"net"
-	"net/http"
 	"net/url"
 	"os"
-	"os/exec"
-	"path/filepath"
-	"strings"
-	"time"
 
 	"github.com/pkg/errors"
-	"github.com/rancher/wrangler/pkg/randomtoken"
+	"github.com/rancher/rancher/pkg/settings"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/crypto/ssh/agent"
 	corev1 "k8s.io/api/core/v1"
-	k8snet "k8s.io/apimachinery/pkg/util/net"
-)
-
-type Options struct {
-	Credential        *corev1.Secret
-	CABundle          []byte
-	InsecureTLSVerify bool
-	Headers           map[string]string
-}
-
-func newGit(directory, url string, opts *Options) (*git, error) {
-	if opts == nil {
-		opts = &Options{}
-	}
 
-	g := &git{
-		URL:               url,
-		Directory:         directory,
-		caBundle:          opts.CABundle,
-		insecureTLSVerify: opts.InsecureTLSVerify,
-		secret:            opts.Credential,
-		headers:           opts.Headers,
-	}
-	return g, g.setCredential(opts.Credential)
-}
+	// gogit packages
+	gogit "github.com/go-git/go-git/v5"
+	config "github.com/go-git/go-git/v5/config"
+	plumbing "github.com/go-git/go-git/v5/plumbing"
+	transport "github.com/go-git/go-git/v5/plumbing/transport"
+	plumbingHTTP "github.com/go-git/go-git/v5/plumbing/transport/http"
+	plumbingSSH "github.com/go-git/go-git/v5/plumbing/transport/ssh"
+)
 
-type git struct {
+// repository holds the config of a git repository and the repo instance.
+type repository struct {
 	URL               string
 	Directory         string
+	username          string
 	password          string
 	agent             *agent.Agent
 	caBundle          []byte
 	insecureTLSVerify bool
 	secret            *corev1.Secret
 	headers           map[string]string
 	knownHosts        []byte
+	repoGogit         *gogit.Repository
+	auth              transport.AuthMethod // interface
```

Comment:
I don't think this comment here is very useful

Reply:
Agreed, removing